aws/rust-runtime/aws-config/src/provider_config.rs

```diff
@@ -285,6 +320,25 @@ impl ProviderConfig {
         self.with_region(provider_chain.region().await)
     }
 
+    /// Use the [default use_fips provider](crate::default_provider::use_fips::use_fips_provider) to set the
+    /// `use_fips` setting for this configuration
+    ///
+    /// Note: the `env` and `fs` already set on this provider will be used when loading the default region.
+    pub async fn load_default_use_fips(self) -> Self {
+        let use_fips = crate::default_provider::use_fips::use_fips_provider(&self).await;
+        self.with_use_fips(use_fips)
+    }
+
+    /// Use the [default use_dual_stack provider](crate::default_provider::use_dual_stack::use_dual_stack_provider) to set the
+    /// `use_dual_stack` setting for this configuration
+    ///
+    /// Note: the `env` and `fs` already set on this provider will be used when loading the default region.
+    pub async fn load_default_use_dual_stack(self) -> Self {
+        let use_dual_stack =
+            crate::default_provider::use_dual_stack::use_dual_stack_provider(&self).await;
+        self.with_use_dual_stack(use_dual_stack)
+    }
```

Comment:
I think doing it this way increases the chances for bugs a little bit. In an ideal world we would:
- make the use_fips/use_dualstack accessors async and have them fallback to the default providers
- call provider_config.use_dual_stacl() methods from the STS constructor. We also need to audit

But that's not going to work well in practice because current callers of `sts_client_config()` don't have the ability to make async calls. So instead, here's what I'm thinking:
Tenets:
1. Customers should never need to call these methods, it should always happen automatically. I think this is what people would expect.
2. It should be hard/impossible to write code that fails to load these values.

Add something like:
```rust
struct ProviderConfig { initialized: bool, ... }

async fn initialize_dependent_fields(&mut self) {
  if self.region == None {
    self.region = region_provider.await;
  }   
  if self.use_dual_stack == None {
    self.use_dual_stack = crate::default_provider::use_dual_stack::use_dual_stack_provider(&self).await;
  }
  initialized = true;
  // etc.
}

// debug asserts so that in tests, we can ensure that all code paths call the initializer.
pub fn use_dual_stack(&self) { if self.use_dual_stack.is_none() { debug_assert!(self.initialized, "must call initialize default fields"); } }
```

Then we call this method in the following places:
- `async fn load()`
- Anywhere else that initializes an AWS client from provider config directly. (e.g. SSO?)

One extremely bulletproof way to do this would be to add a generic to provider config, `Initialized` and only accept an initialized one in these places? Food for though.